adapters/repos/db/vector/flat/index_test.go

```diff
@@ -179,7 +200,44 @@ func Test_NoRaceFlatIndex(t *testing.T) {
 	})
 
 	t.Run("recall on compression with deletes", func(t *testing.T) {
-		recall, latency, err := run(dirName, logger, flatent.CompressionBQ, vectors, queries, k, truths, extraVectorsForDelete, nil, distancer)
+		recall, latency, err := run(dirName, logger, CompressionBQ, false, vectors, queries, k, truths, extraVectorsForDelete, nil, distancer)
+		require.Nil(t, err)
+
+		fmt.Println(recall, latency)
+		assert.Greater(t, recall, float32(0.8))
+		assert.Less(t, latency, float32(1_000_000))
+	})
+
+	t.Run("recall on no compression with cache", func(t *testing.T) {
```

Comment:
if the only difference between previous and newly added tests is cache being turned on/off, tests duplication can be avoided by moving them to the loop with parametrised `cache` value:

```
for _, cache := range []bool{false, true} {
	t.Run("cache: "+strconv.FormatBool(true), func(t *testing.T) {
		t.Run("test1", func(t *testing.T) {
			run(cache, ...)
		})
		t.Run("test2", func(t *testing.T) {
			run(cache, ...)
		})
	})
}
```